kngsild/src/main/kotlin/io/egm/kngsild/utils/AuthUtils.kt

```diff
@@ -14,52 +14,76 @@ import java.net.URI
 import java.net.http.HttpRequest
 import java.net.http.HttpResponse
 
-class AuthUtils(
-    private val serverUrl: String,
-    private val clientId: String,
-    private val clientSecret: String,
-    private val grantType: String,
-    private val enabled: Boolean = true
-) {
+class AuthUtils {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
+    private val accessToken: String?
+
+    private val serverUrl: String?
+    private val clientId: String?
+    private val clientSecret: String?
+    private val grantType: String?
+    private val enabled: Boolean?
+
+    constructor(accessToken: String) {
+        this.accessToken = accessToken
+        this.serverUrl = null
+        this.clientId = null
+        this.clientSecret = null
+        this.grantType = null
+        this.enabled = null
```

Comment:
you should handle the `enabled` info as before or better add an `AuthType` enum (`None`, `ProvidedToken`, `ClientCredentials`)?

Reply:
yes I do something like that